### ISSUE TITLE
FEAT - WeightedLogistic datafit

### ISSUE THREAD

**Issue by carlosg-m** [scikit-learn-contrib/skglm#335]
## Description of the feature

Support sample weights in Logistic Regression just like sklearn.

**Comment by mathurinm** [comment]
@Badr-MOUFAD this should not be too hard to do if you want to revive it :) updating gradient and Hessian basically ?
